src/v/pandaproxy/client/brokers.cc

```diff
@@ -15,8 +15,7 @@ namespace pandaproxy::client {
 
 ss::future<> brokers::stop() {
     return ss::parallel_for_each(
-      std::move(_brokers),
```

Comment:
Please explain in the commit message why the brokers shouldn't be moved.

Reply:
@BenPope all of seastar types only take a ref-type, they do not take a value type. so i'd also check your for_each, etc. all looping constructs just take a ref type.

Reply:
`_brokers may be used on the callback from outstanding requests.` this feels like there is a missing synchronization step here. since the `std::move(_brokers)` should leave that data structure in a valid (but empty) state, if something is racing with `brokers::stop` then your comment seems to suggest that the race depends on stopped brokers being in that structure? i think a bit more explanation might be useful.

Reply:
Maybe the move was fine. It's hard to repro. I can try without it.

Reply:
> @BenPope all of seastar types only take a ref-type, they do not take a value type. so i'd also check your for_each, etc. all looping constructs just take a ref type.

One idea I was experimenting with for `parallel_for_each` was to also combine the equivalent of an `ss::do_with`, and `move_iterators` if the container was an rvalue ref.

Reply:
> `_brokers may be used on the callback from outstanding requests.` this feels like there is a missing synchronization step here. since the `std::move(_brokers)` should leave that data structure in a valid (but empty) state, if something is racing with `brokers::stop` then your comment seems to suggest that the race depends on stopped brokers being in that structure? i think a bit more explanation might be useful.

The move was safe, the commit has been dropped .